ext/standard/tests/strings/sprintf_rope_optimization_003.phpt

```diff
@@ -0,0 +1,134 @@
+--TEST--
+Test sprintf() function : Rope Optimization for '%d'.
+--FILE--
+<?php
+function func($num) {
+	return $num + 1;
+}
+function sideeffect() {
+	echo "Called!\n";
+	return "foo";
+}
+class Foo {
+	public function __construct() {
+		echo "Called\n";
+	}
+}
```

Comment:
It might be interesting to have a test (possibly in a different file) with a GMP object being cast to int.

Reply:
Added. It just works (identically for optimization and no optimization).